### PR TITLE
Extract nginx to a separate chart

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**

Changes proposed in this pull request:

- application-connector nginx-ingress-controller extracted to a separate chart: `application-connector-ingress`
- nginx-ingress service (type: LoadBalancer) labelled with: `application-connector-lb: "true"` label.
- added migration job to enable upgrade from release 0.7 by resolving conflict on LoadBalancers and a single public IP Address
- renamed secret defined in application-connector chart from `tls-ingress-cert` to `application-connector-tls-ingress-cert` to distinguish from another secret with the same name defined in `cluster-essentials`

**Related issue(s)**
`See also #2434 `.

### CODE REVIEW

File: resources/application-connector/templates/nginx-ingress-cert.yaml [hunk 1]
```diff
@@ -4,6 +4,6 @@ data:
...ClJhVStpSGJqTE13d3YwWWpDZXlXWk53bFNDUWc5U2g4bmVjMWpidXBMOSt2a2VXM1ZvWjNLSXBCWGhzME8xL0gKcmpnZjdSMWhROXExK0kyY1pzdmtPK0JZcnk4UGszSW9MS29rVnF1RVM3d2w0UmVFTThQNU0vV2VzRzNkd1grYwpHalkxSS90YjFLYlJBWmtma1JsZDFPZ1FhUUtCZ1FEQWVXOVVQaU9zYmY1clV1dkF0WFk0YXRBdkZlTmxCYmFOClgvZnNRa2h3UVZwWlJadFVKMGtwWWRlbFpqdnk5VVJ5Tk40RUl6bnF6VXYyakhYb3ZrajNMR3NIcHBUbzdyRkMKZUptZDU5RFB0cHNFZVh4My93RXMyVTBGUGlWU1BGOXMzemNPWHU0VGhnS1N5dkUrd0FJMXU0Ym84d3FIMFhZMgphZS8wTWJRUzR3S0JnSHVxVzBjVFpzcDdldTdvbVhCdUlNUm5YTjhMdUtFNUs5cjZMVERkL1V6YVdHQXlHcG80CmNCWFFtVkF5YXByR3k3d0lFUWphU3JyL212cnhuc2huc1BkTSs1WUl4eDZTYXN3UzZmRk83TmhHWklXbHM5d3QKNlYvdHdOdG9jM3Exa2c2S2ZvTGpaSlpRMENoMkU3VEJQcGlKemYzUjcvYk5JdkhxQjlZb3Vsb0JBb0dBV25lNQprRXMwMFZOU0JuZ1BQNkVMVkRGQWNTRkVwTE11dnZ5Z3QzZmdQSU03U0VLalk4YWphTVFJYkVIRlk4bE14ckY2CnVVOGxaZXZoREoyd3ZoalNpdnRodzFMVkl6Y3VNaVFsY0VjSnF0Z1Z2T0N0VkdtVHo5VERrdmRHNjRSZEJmdFYKRVhnbFJ4L3lEYlU5OTFTZ3RValRmS0xnREQ2ejVaZWRwWlhISWlrQ2dZRUFoL0lRMnhuS1hOUUZlL0d6cU1sYQpZZ3NqaWxhSkJlcW1qOTd6S2h5a3JvclZCTFhnU0ErSU5pMGpXY2pPUVZ5RmxNcjhBRFIvRm01NWNLTFFSOCtLCjNUOGFFUjJRRExTVVh5UWtkS0w0VVAvKzBXRmpSdGF6cktuV1JHZkl6Vmk2L253RXE2eVo4N3JQZGdwRmQzQmoKS0U0L1k0MFBSa21xY2wvc0FoUmdkK1U9Ci0tLS0tRU5EIFBSSVZBVEUgS0VZLS0tLS0K" }}
```
Comment (Demonsthere) [inline resources/application-connector/templates/nginx-ingress-cert.yaml]: Didn't we want to delete this deprecated value?

Comment (Tomasz-Smelcerz-SAP) [reply resources/application-connector/templates/nginx-ingress-cert.yaml]: Removed.
Reason: This certificate was old, issued for a domain we don't use anymore.
A valid "default" is already set in our installation scripts, so a default value in Helm charts will not be used anyway (it's always set)

File: resources/application-connector-nginx/templates/upgrade-job.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      restartPolicy: OnFailure
+      containers:
+      - name: {{ .Release.Name }}-loadbalancer-migration
+        image: eu.gcr.io/kyma-project/develop/xip-patch:1103850e
+        command:
+          - bash
+          - -c
+          - |
+            set -e
+            SERVICE_TO_REMOVE="application-connector-nginx-ingress-controller"
```
Comment (Szymongib) [inline resources/application-connector-nginx/templates/upgrade-job.yaml]: Shouldn't it be the name of the service? If so it is not accurate as the service name changed to `application-connector-nginx-nginx-ingress-controller`

Comment (Tomasz-Smelcerz-SAP) [reply resources/application-connector-nginx/templates/upgrade-job.yaml]: This is OK. We have to remove old service here - and it is the old name. This job is only for migration from "old" (0.7) to "new" (0.8) and shouldn't be used for anything else, hence the name is hardcoded.

File: resources/application-connector/templates/nginx-ingress-cert.yaml [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
+  "tls.crt": {{ .Values.global.tlsCrt }}
+  "tls.key": {{ .Values.global.tlsKey }}
 kind: Secret
 metadata:
-  name: ingress-tls-cert
+  name: application-connector-ingress-tls-cert
```
[the remark this answers is not in the document]
Comment (Tomasz-Smelcerz-SAP) [reply resources/application-connector/templates/nginx-ingress-cert.yaml]: Addressed in a separate PR: #2969

File: installation/resources/installer-cr-cluster.yaml.tpl [hunk 4]
```diff
@@ -36,8 +36,10 @@ spec:
       namespace: "kyma-system"
     - name: "event-bus"
       namespace: "kyma-system"
-    - name: "application-connector"
+    - name: "application-connector-nginx"
```
[the remark this answers is not in the document]
Comment (Tomasz-Smelcerz-SAP) [reply installation/resources/installer-cr-cluster.yaml.tpl]: I agree. Please propose another name :)

File: components/xip-patch/delete-loadbalancer.sh [hunk 5]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
```
Comment (Tomasz-Smelcerz-SAP) [inline components/xip-patch/delete-loadbalancer.sh]: Not used, remove

File: installation/resources/installer-cr-cluster.yaml.tpl (hunk 4 above)
[the remark this answers is not in the document]
Comment (Szymongib) [reply installation/resources/installer-cr-cluster.yaml.tpl]: After some discussion, we decided that the best name would be `application-connector-ingress`

File: components/xip-patch/delete-loadbalancer.sh (hunk 5 above)
Comment (Tomasz-Smelcerz-SAP) [reply components/xip-patch/delete-loadbalancer.sh]: Removed

File: installation/resources/installer-cr-cluster.yaml.tpl (hunk 4 above)
[the remark this answers is not in the document]
Comment (Tomasz-Smelcerz-SAP) [reply installation/resources/installer-cr-cluster.yaml.tpl]: Renamed :)

**Pull request merged**